### PR TITLE
Remove peerNodeId in MCSP receive table (Fix: #6340)

### LINKED ISSUE #6340: Add another entry point for MSCP received message re-entrance.
#### Problem
 Currently `peerNodeId`, `packetHeader`, `peerAddress`, `msgBuf` is stored for MSCP received message, which is delivered once MCSP is complemented. We may omit `packetHeader`, `peerAddress` fields by refactoring secure session manager, and add another entry point of `OnMessageReceived` which it only take `msgBuf` as argument.

### PR DESCRIPTION
**Pull request by kghost**
Fixes: #6340

### CODE REVIEW

**Comment by kghost** [comment]
This PR is trying to reduce memory usage of MCSP retrans table in bss section. But ends up using more memory.

.text increased by function `MessageCounterManager::ProcessPendingMessages`, which is decoding the packet header.
```
  78    106    184   chip::secure_channel::MessageCounterManager::ProcessPendingMessages(unsigned long long)
```

.rodata is probably increased by the error log string:

> MessageCounterManager::ProcessPendingMessages: Failed to decode PacketHeader

File: src/protocols/secure_channel/MessageCounterManager.h [hunk 1]
```diff
@@ -89,10 +90,8 @@ class MessageCounterManager : public Messaging::ExchangeDelegate, public Transpo
      */
     struct ReceiveTableEntry
     {
-        ReceiveTableEntry() : peerNodeId(kUndefinedNodeId) {}
+        ReceiveTableEntry() {}
```
Comment (andy31415) [inline src/protocols/secure_channel/MessageCounterManager.h:93]: can this line be deleted alltogether?

**Review by andy31415 — APPROVED** [review]

**Review by jelderton — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

**Comment by kghost** [comment]
qpg is encountering memory problem again. I'll rebase this PR over #6874 once it is merge, #6874 will free up around 1K mem.

**Pull request merged**